File: TransCoder_tokenized_test_set_functions/java/CHECK_ARRAY_CONTAINS_CONTIGUOUS_INTEGERS_DUPLICATES_ALLOWED.java
static boolean areElementsContiguous ( int arr [ ], int n ) {
  int max = Integer . MIN_VALUE;
  int min = Integer . MAX_VALUE;
  for ( int i = 0;
  i < n;
  i ++ ) {
    max = Math . max ( max, arr [ i ] );
    min = Math . min ( min, arr [ i ] );
  }
  int m = max - min + 1;
  if ( m > n ) return false;
  boolean visited [ ] = new boolean [ n ];
  for ( int i = 0;
  i < n;
  i ++ ) visited [ arr [ i ] - min ] = true;
  for ( int i = 0;
  i < m;
  i ++ ) if ( visited [ i ] == false ) return false;
  return true;
}

File: results/py/CHECK_ARRAY_CONTAINS_CONTIGUOUS_INTEGERS_DUPLICATES_ALLOWED.cpp
bool f_filled ( int * arr , int n ) {
  int max1 = max ( arr ) ;
  int min1 = min ( arr ) ;
  int m = max1 - min1 + 1 ;
  if ( ( m > n ) || ( m < 0 ) ) return false ;
  bool visited [ m ] ;
  memset ( visited , 0 , m ) ;
  for ( int i = 0 ;
  i <= n ;
  i ++ ) visited [ arr [ i ] - min1 ] = true ;
  for ( int i = 0 ;
  i <= m ;
  i ++ ) {
    if ( ( visited [ i ] == false ) == false ) return false ;
  }
  return true ;
}
